### PR TITLE
Chart bug - The orange order, the new one is no longer shown

### LINKED ISSUE #217: Price-Volume chart is not correctly adjusted when change price-volume
When I select a price greater than the last value on the chart, new orders lines is moves outside the chart's area. 
Moreover, when then change the price, new orders line stops to be displayed at all. See the video: https://drive.google.com/file/d/1snhjAl2Dmp7u5N1ULZLkH_FuozXTZDVV/view

### LINKED ISSUE #224: Chart should show Bid's horizontal line go until Y axis.
The bid's horizontal line should go all the way to the Y axis, as shown below:

Screenshot 2021-03-18 at 17 31 24

The area of the bid rectangle should also have a green shade.

### LINKED ISSUE #228: Chart Price and Volume show wrong token
Example taken from [Auction 33](https://ido-ux.dev.gnosisdev.com/#/auction?auctionId=33&chainId=4#topAnchor)

The chart shows price (x) as auctioneer token (WETH), when it should be the bid token(DAI). 

The hover-on popup show's volume in auctioneer token (WETH) instead of bid token(DAI). 

Both are marked below, and should be DAI instead of WETH:

Screenshot 2021-03-18 at 23 51 52

Why? 

This is because in Gnosis Auction bidders determine what amount of bid (DAI) token they want to sell at what price (expressed in bid token again), for a quantity of auctioneer token (WETH) they yet don't know.

### LINKED ISSUE #258: Number of decimals in the chart
image
not more than 2 decimals should be shown

### LINKED ISSUE #301: Chart bug -  The orange order, the new one  is no longer shown
See comment https://github.com/gnosis/ido-ux/pull/297#pullrequestreview-620186754

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #301 

Also:
closes #228 (see last comment here)
closes #224 
closes #258 
closes #217 

I removed the gridlines added manually, because breaks the chart in some cases

Look like this ...

Screenshot_20210401_004316

### CODE REVIEW

File: src/components/auction/OrderbookWidget/index.tsx [hunk 1]
```diff
@@ -140,8 +140,8 @@ const processData = (
         // Data for representation
         priceNumber: price,
         totalVolumeNumber: totalVolume,
-        priceFormatted: price.toString(),
-        totalVolumeFormatted: totalVolume.toString(),
+        priceFormatted: price.toFixed(2),
```
Comment (josojo) [inline src/components/auction/OrderbookWidget/index.tsx:143]: Yes, this is a nice fix. I think I would prefer toSignificant(6), though.

File: src/hooks/useChart.ts [hunk 2]
```diff
@@ -1,46 +1,59 @@
...
-  const elemRef = useRef(null)
+}
+
+const useChart = (props: Props) => {
+  const { baseToken, createChart, data, quoteToken } = props
+
   const [loading, setLoading] = useState(false)
 
+  const mountPoint = useRef<Maybe<HTMLDivElement>>(null)
+  const chartRef = useRef<Maybe<am4charts.XYChart>>(null)
```
Comment (josojo) [inline src/hooks/useChart.ts:22]: Could you please leave a comment in the code that explains this mountPoint logic. 

I would love to understand your solution, given that it was holding us back so long.

**Comment by josojo** [comment]
When I was testing the PR, my computer got loud again, the ventilator needed to work again. 
I think we still have the performance problem in there...

Sorrry that this is soo hard.

**Comment by elena-zh** [comment]
HI @mariano-aguero , I have reviewed the changes. The chart looks much better now, however, here are several notes: 
1. when I create an auction, and there are no orders placed, a green line with 0 volume and 'strange' price is displayed in the chart.


2. then, I see a red line going up with also 'unknown' price and volume. Price value increases when I place orders. Is that OK?

Here the video with the cases 1-2: https://drive.google.com/file/d/1rd0Ls8VitWlu2PiYbUfpPrFvAjLgk2zK/view

3. #217 issue seems to be unfixed: [an empty chart is displayed when I enter a high price](https://drive.google.com/file/d/1y7aqQ4FfYimBU_HRYDLgIBj33uhjM_LI/view), then, at any moment, orange area stops to be displayed when I try to build an order. Moreover, strange 'minus' icon  and DAI volume in million appear when I play with 'orange' area and try to place an order ([video](https://drive.google.com/file/d/1y7aqQ4FfYimBU_HRYDLgIBj33uhjM_LI/view))


Could you please take a look into these issues?

File: src/components/auction/OrderbookWidget/index.tsx (hunk 1 above)
Comment (mariano-aguero) [reply src/components/auction/OrderbookWidget/index.tsx:143]: Take into account the variable is a number, not a Fraction object

File: src/components/auction/Charts/XYChart.ts [hunk 3]
```diff
@@ -61,76 +61,42 @@ const XYChart = (args) => {
...
+  bidSeries.startLocation = 0.5
+  bidSeries.fillOpacity = 0.1
   bidSeries.dummyData = {
     description:
       'Shows the price (x axis) and size (y axis) of the bids that have been placed, both expressed in the bid token',
   }
-  bidSeries.tooltipText = `[bold]${market}[/]\nBid Price: [bold]{priceFormatted}[/] ${quoteTokenLabel}\nVolume: [bold]{totalVolumeFormatted}[/] ${quoteTokenLabel}`
 
   // Create serie, red line, shows the minimum sell price (x axis) the auctioneer is willing to accept
-  const askSeries = chart.series.push(new am4charts.LineSeries())
```
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/components/auction/Charts/XYChart.ts:117]: My bad, I will revert this

File: src/hooks/useChart.ts (hunk 2 above)
Comment (mariano-aguero) [reply src/hooks/useChart.ts:22]: Sure

**Comment by mariano-aguero** [comment]
> When I was testing the PR, my computer got loud again, the ventilator needed to work again.
> I think we still have the performance problem in there...
> 
> Sorrry that this is soo hard.

I think this is the best version of the chart we have since the beginning of the application. In the gif we can see that we modify the order and there is no impact for the user (blocking the input of numbers at some point when performing some action with the keyboard)

Peek 01-04-2021 08-38

We can also see that the performance has improved in relation to previous versions, is more smooth.
Screenshot_20210401_084231

**Comment by mariano-aguero** [comment]
> HI @mariano-aguero , I have reviewed the changes. The chart looks much better now, however, here are several notes:
> 
> 1. when I create an auction, and there are no orders placed, a green line with 0 volume and 'strange' price is displayed in the chart.
>    
>    
> 2. then, I see a red line going up with also 'unknown' price and volume. Price value increases when I place orders. Is that OK?
>    
>    Here the video with the cases 1-2: https://drive.google.com/file/d/1rd0Ls8VitWlu2PiYbUfpPrFvAjLgk2zK/view
> 3. #217 issue seems to be unfixed: [an empty chart is displayed when I enter a high price](https://drive.google.com/file/d/1y7aqQ4FfYimBU_HRYDLgIBj33uhjM_LI/view), then, at any moment, orange area stops to be displayed when I try to build an order. Moreover, strange 'minus' icon  and DAI volume in million appear when I play with 'orange' area and try to place an order ([video](https://drive.google.com/file/d/1y7aqQ4FfYimBU_HRYDLgIBj33uhjM_LI/view))
>    
> 
> Could you please take a look into these issues?

I will check the points 1 and 2, I think the point 3 is an edge case and take into consideration the proportions of the chart and this issue https://github.com/gnosis/ido-ux/issues/262 , nothing will work

**Comment by mariano-aguero** [comment]
> HI @mariano-aguero , I have reviewed the changes. The chart looks much better now, however, here are several notes:
> 
> 1. when I create an auction, and there are no orders placed, a green line with 0 volume and 'strange' price is displayed in the chart.
>    
>    
> 2. then, I see a red line going up with also 'unknown' price and volume. Price value increases when I place orders. Is that OK?
>    
>    Here the video with the cases 1-2: https://drive.google.com/file/d/1rd0Ls8VitWlu2PiYbUfpPrFvAjLgk2zK/view
> 3. #217 issue seems to be unfixed: [an empty chart is displayed when I enter a high price](https://drive.google.com/file/d/1y7aqQ4FfYimBU_HRYDLgIBj33uhjM_LI/view), then, at any moment, orange area stops to be displayed when I try to build an order. Moreover, strange 'minus' icon  and DAI volume in million appear when I play with 'orange' area and try to place an order ([video](https://drive.google.com/file/d/1y7aqQ4FfYimBU_HRYDLgIBj33uhjM_LI/view))
>    
> 
> Could you please take a look into these issues?

I checked the point #1, this situation already exist in the develop branch, so having said that and not being an "error" introduced by this PR it should not be blocking. Seems to be a problem  with "how it generates the data array with the information" .

Develop
Screenshot_20210401_093444

This PR
Screenshot_20210401_093729


For point 2, I think it should work like this since adding that order to the chart shows how it would vary

For the point 3, appears to be an edge case that differs from the issue #217 in how the application is used(adding a huge  number) See #262 for more info too

**Review by gabitoesmiapodo — APPROVED** [review]
Some small(ish) issues notwithstanding, I detect no lag nor laptop fans going crazy on my side.

The orange line is there. It can be difficult to see if the user enters a too small or too big amount, but it's there.

Also, all fields on the form must be completed for the orange line to appear. That's expected, I think.

Some of the other (general) performance issues seem to be related to the app architecture, and maybe the use of Redux too, but I'm mostly using my intuition (not a really deep technical analysis) here.

**Comment by elena-zh** [comment]
Yes, the issue #376 was closely about the issue described in the case 1. The only thing that is was applied on the 1st order, but still, green line for an auction without orders presents in Dev

**Pull request merged**